packages/backend/server/src/core/notification/service.ts

```diff
@@ -77,21 +83,50 @@ export class NotificationService {
   }
 
   async createInvitation(input: InvitationNotificationCreate) {
-    const isActive = await this.models.workspaceUser.getActive(
-      input.body.workspaceId,
-      input.userId
-    );
-    if (isActive) {
-      this.logger.debug(
-        `User ${input.userId} is already a active member of workspace ${input.body.workspaceId}, skip creating notification`
-      );
+    const workspaceId = input.body.workspaceId;
+    const userId = input.userId;
+    if (await this.isActiveWorkspaceUser(workspaceId, userId)) {
       return;
     }
-    await this.ensureWorkspaceContentExists(input.body.workspaceId);
-    return await this.models.notification.createInvitation(
+    await this.ensureWorkspaceContentExists(workspaceId);
+    const notification = await this.models.notification.createInvitation(
       input,
       NotificationType.Invitation
     );
+    await this.sendInvitationEmail(input);
+    return notification;
+  }
+
+  private async sendInvitationEmail(input: InvitationNotificationCreate) {
+    const inviteUrl = this.url.link(`/invite/${input.body.inviteId}`);
+    if (this.config.node.dev) {
+      // make it easier to test in dev mode
+      this.logger.debug(`Invite link: ${inviteUrl}`);
+    }
+    const userSetting = await this.models.settings.get(input.userId);
+    if (!userSetting.receiveInvitationEmail) {
+      return;
+    }
+    const receiver = await this.models.user.getWorkspaceUser(input.userId);
```

Comment:
if I remember correctly, the caller know every arguments of notification will use, `user`, `invitation`. It's a directly function call, so avoid to refetch them all

Reply:
Triggered by the `notification.sendInvitation` job, the call chain has not retrieved `user` information before, this is the first time it is being retrieved here.